### DISCUSSION TITLE
Use untyped pointers for wgpu-hal raw Vulkan API?

### DISCUSSION THREAD

**Discussion by zmerp** [gfx-rs/wgpu#2615] [Ideas]
Reading this comment [here](https://github.com/gfx-rs/wgpu/issues/2592#issuecomment-1099553745) I think it would be desirable to not expose the ash crate in the public interface and instead use raw pointers. This approach was already used [here](https://github.com/gfx-rs/wgpu/blob/master/wgpu/src/lib.rs#L1511).

Any opinion from wgpu's maintainers?

**Comment by cwfitzgerald** [comment]
If we could do it in a way that is minimally invasive, I think that would be totally fine -- it would also allow for the users to use something other than ash if they so desired.

**Reply by zmerp** [reply]
Great. I could make a PR in the near future.

**Reply by kvark** [reply]
Yes, assuming `ash` has its own version of `from_raw`, which we'd need to call on raw data.
Would be great to make it a private dependency again.

**Reply by MarijnS95** [reply]
> it would also allow for the users to use something other than ash if they so desired.

Fwiw it'd be great if Rust could just have one set of "official" Vulkan bindings, just like how Khronos has an official set of headers for C and C++, everyone else just builds on top.

**Reply by zmerp** [reply]
ash would be a good candidate for that, but the problem here is that ash and wgpu often go out of sync and wgpu forces the user to use and older ash version.

**Reply by MarijnS95** [reply]
To give some context:

1. This many breaking release-versions - with extremely minor actual breaking changes - in such short succession wasn't my intent, and it'll now take quite a while before we do it again (there will still be additive patch releases though!);
2. The MSRV bump also wasn't intentional, and we now check this in the CI to actively reject - or postpone - PRs that require aggressive bumps.

In other words, if you're making these changes specifically to address those points, I wouldn't bother.
